ablog/__init__.py

```diff
@@ -92,15 +95,13 @@ def setup(app):
         latex=(lambda s, n: s.visit_admonition(n), lambda s, n: s.depart_admonition(n)),
     )
 
-    pkgdir = os.path.abspath(os.path.dirname(__file__))
-    locale_dir = os.path.join(pkgdir, "locales")
-    app.config.locale_dirs.append(locale_dir)
+    app.config.locale_dirs.append(os.path.join(PKGDIR, "locales"))
 
     return {"version": __version__}  # identifies the version of our extension
 
 
 def config_inited(app, config):
-    app.config.templates_path.append(get_html_templates_path())
+    app.config.templates_path = [get_html_templates_path()] + app.config.templates_path
```

Comment:
This should prevent ablog being last on the list?

Reply:
This looks... wrong. This puts ablog *before* the user-configured templates, which means even the user can't override these files. These are placed before theme templates too, so themes still won't be able to override the newly moved files.

Reply:
Ok, so I don't want to add ablog templates to this at all?

Reply:
This code has some fairly useful properties and behaviours:

- All the attributes being used here are public, so Sphinx's backwards compatibility policy applies on them, giving us ample time to make changes if Sphinx changes things (deprecation warnings for 2 major versions before removal, which should be more than an year to adapt this code)
- This puts the ablog templates *after* the theme templates, by default, with an escape hatch for users who don't want that. This does depend on them not setting `html_template_bridge` to something super custom -- that's a very small set of users AFAICT and the error message should give such users all the information that they need to act on it.
- This allows themes to *change* where ablog's template injection occurs, by setting a flag in their `theme.conf` file:

  ```
  [ablog]
  inject_templates_after_theme = true
  ```

  This would enable themes like Furo to contain their own `ablog/catalog.html` for
  example.

In the spirit of transparency, I've not tested this code in any way -- and written most of this in a Markdown editor -- so no promises on whether this will work. That said, this will enable basically everything we discussed in #143 and #125 (as long as you move the templates, as has been done already). This should also be a drop-in replacement for users who don't set up custom templates or overrides on top of ablog itself. If they do, they might need to move those paths to match the namespace.

Reply:
Thanks for all of this, still trying to work out the kinks but your code really helped!